feat: Release `1.0.0-pre.9` (#3579)
CloudFire-LLC/cloudfire: website/src/app/kb/deploy/sites/readme.mdx

import PlanBadge from "@/components/PlanBadge";
import SupportOptions from "@/components/SupportOptions";
import NextStep from "@/components/NextStep";

<PlanBadge plans={["starter", "enterprise"]}>

# Sites

</PlanBadge>

Sites represent a shared network environment that gateways and resources exist
within. All gateways and resources in a site need unobstructed network
connectivity to each other. This is typically achieved by placing all resources
and gateways in the same physical location or virtual network.

## What is a site?

A site is a collection of resources and gateways that are logically grouped
together. A site can be a physical location, a virtual network, or any other
grouping that makes sense for your organization.

A site can be as small as a single gateway running on a single server managing
the access to the resources on that server, or as large as an entire data center

> managing access to an entire fleet of servers.

Reviewer: I think the data center will not have a network environment where everything can access everything, maybe we should align the site explanation closer to VPC or some DevOps term well understood by our users?

Author: Sure, a better description here would be "data center rack"

Author: Updated to say "data center network managing access to an entire cluster of servers". This will be common for AI workloads.

import PlanBadge from "@/components/PlanBadge";
import SupportOptions from "@/components/SupportOptions";
import NextStep from "@/components/NextStep";
import Alert from "@/components/DocsAlert";

<PlanBadge plans={["starter", "enterprise"]}>

# Sites

</PlanBadge>

Sites represent a shared network environment that Gateways and Resources exist

> within. All Gateways and Resources in a Site must have **unobstructed network
> connectivity** to each other.

Reviewer 2: ```suggestion
within. All Gateways in a Site must have **unobstructed network
connectivity** to all Resources in the Site.
```

The Gateways never connect to other Gateways and we don't need Resources to connect to each other since our scope is only that Gateways connect to Resources, right? Resources don't even need to be able to connect to Gateways if there's a trapdoor firewall for some reason.

## What is a site?

A site is a collection of resources and gateways that are logically grouped
together. A site can be a physical office location, a data center rack, a
homelab network, VPC, or any other grouping that makes sense for your
organization.

A site can be as small as a single gateway running on a single server managing

> the access to the resources on that server, or as large as an entire data center

Reviewer 2: This answers my question from the other comment.

> network managing access to an entire cluster of servers.

Reviewer 3: ```suggestion
network managing access to multiple clusters of servers.
```

The use of `entire` twice in one sentence just felt unusual to me.  Just a personal preference.

> ## Creating a site

Reviewer 3: I'll let this comment stand in for all places in the docs where this is the same...

Should we capitalize things like `Site`, `Resource`, `Gateway`, etc... regardless of where they occur in the sentence, since they refer to specific types within our system?  I feel like I've seen this in other software documentation and just wanted to throw it out as a suggestion since a lot of our terminology within our system is close to generic terminology.

Author: Yeah, I think it makes sense to capitalize them all whenever they're referring to Firezone entities.

For example, "Create a Resource", but not "Firezone secures access to your company's most sensitive resources".

Reviewer 3: Yeah, that was exactly how I was thinking about it as well.

> managing access to an cluster of servers.

Reviewer 2: ```suggestion
managing access to a cluster of servers.
```
